runtime/vm/createramclass.cpp

```diff
@@ -1604,19 +1609,25 @@ loadSuperClassAndInterfaces(J9VMThread *vmThread, J9ClassLoader *classLoader, J9
 			}
 		}
 	}
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	{
+		J9UTF8 *nestHostName = J9ROMCLASS_NESTHOSTNAME(romClass);
 
-	return TRUE;
-}
+		/* Only loads nest host if one is named*/
+		if (NULL != nestHostName) {
+			*nestHostOut = internalFindClassUTF8(vmThread, J9UTF8_DATA(nestHostName), J9UTF8_LENGTH(nestHostName), classLoader, classPreloadFlags);
```

Comment:
If the nest host is not visible to the class being loaded does Oracle?

If so we will need to do a visibility check here before assigning to nestHostOut